eclair-core/src/test/scala/fr/acinq/eclair/channel/states/h/ClosingStateSpec.scala

```diff
@@ -528,7 +528,7 @@ class ClosingStateSpec extends TestKitBaseClass with FixtureAnyFunSuiteLike with
     // note that bob doesn't receive the new sig!
     // then we make alice unilaterally close the channel
     val closingState = localClose(alice, alice2blockchain)
-    assert(closingState.commitTx === aliceCommitTx)
+    assert(closingState.commitTx.txid === aliceCommitTx.txid)
```

Comment:
I think we need to test somewhere (not necessarily here, I'm not sure which test is the best place for that), that the commit txs sent to the `alice2blockchain`/`bob2blockchain` are fully signed. We don't need to repeat this check in every test, matching on the `txid` is ok in most places, but we need to test the codepaths where we're supposed to sign the commit tx (and that applies to htlc txs as well).

Reply:
I asked myself the same question. A few lines above, we could totally do:
```scala
val aliceCommitTx = alice.stateData.asInstanceOf[DATA_NORMAL].commitments.fullySignedLocalCommitTx(alice.underlyingActor.nodeParams.channelKeyManager)
```

And then keep the full equality check. This is what we do in the `TxPublisher` test. That said, in the state tests it wouldn't actually check that transaction are properly signed, because it would requires an actual bitcoind. We do that in the integration tests.

Reply:
Instead of calling `commitments.fullySignedLocalCommitTx` I would instead grab the commit tx we intercept with the `alice2blockchain` actor, and call `CorrectlySpends(fundingTx)` on it, don't you think?

And as mentioned in another comment. we should verify in some tests that what we store is unsigned.

Reply:
Done with 6f96eae.